helita/sim/ebysus.py

```diff
@@ -16,48 +17,66 @@ class EbysusData(BifrostData):
+        self.snapevars = ['e']
         self.mhdvars = []
         if (self.do_mhd):
             self.mhdvars = ['bx', 'by', 'bz']
-        self.auxvars = self.params['aux'].split()
+        self.auxvars = self.params['aux'][self.snapInd].split()
+
+        self.compvars = ['ux', 'uy', 'uz', 's', 'ee']
+
+        self.varsmfc = [v for v in self.auxvars if v.startswith('mfc_')]
+        self.varsmf = [v for v in self.auxvars if v.startswith('mf_')]
+        self.varsmm = [v for v in self.auxvars if v.startswith('mm_')]
+        self.varsmfe = [v for v in self.auxvars if v.startswith('mfe_')]
 
         if (self.mf_epf):
             # add internal energy to basic snaps
-            self.snapvars.append('e')
+            #self.snapvars.append('e')
             # make distiction between different aux variable
-            self.varsmfe = [v for v in self.auxvars if v.startswith('mfe_')]
-            self.varsmfc = [v for v in self.auxvars if v.startswith('mfc_')]
-            self.varsmf = [v for v in self.auxvars if v.startswith('mf_')]
-            for var in (self.varsmfe + self.varsmfc + self.varsmf):
-                self.auxvars.remove(var)
+            self.mf_e_file = self.file_root + '_mf_e'
         else:  # one energy for all fluid
-            self.mhdvars = 'e' + self.mhdvars
-            if self.with_electrons:
-                self.snapevars.remove('ee')
+            self.mhdvars.insert(0, 'e')
+            self.snapevars = []
+            
         if hasattr(self, 'with_electrons'):
             if self.with_electrons:
                 self.mf_e_file = self.file_root + '_mf_e'
+                # JMS This must be implemented
+                self.snapelvars=['r', 'px', 'py', 'pz', 'e']
+
+        for var in (
+                self.varsmfe +
+                self.varsmfc +
+                self.varsmf +
+                self.varsmm):
+            self.auxvars.remove(var)
+
+        #if hasattr(self, 'mf_total_nlevel'):
+        #    if self.mf_total_nlevel == 1:
+        #        self.snapvars.append('e')
+
+        self.simple_vars = self.snapvars + self.snapevars + self.mhdvars + self.auxvars + \
+            self.varsmf + self.varsmfe + self.varsmfc + self.varsmm
 
         self.auxxyvars = []
         # special case for the ixy1 variable, lives in a separate file
         if 'ixy1' in self.auxvars:
             self.auxvars.remove('ixy1')
             self.auxxyvars.append('ixy1')
 
-        self.simple_vars = self.snapvars + self.mhdvars + \
-            self.auxvars + self.varsmf + self.varsmfe + self.varsmfc
-        self.compvars = ['ux', 'uy', 'uz', 's', 'rup', 'dxdbup',
-                         'dxdbdn', 'dydbup', 'dydbdn', 'dzdbup', 'dzdbdn', 'modp']
+        for var in self.auxvars:
+            if any(i in var for i in ('xy', 'yz', 'xz')):
+                self.auxvars.remove(var)
+                self.vars2d.append(var)
+
+        '''self.compvars = ['ux', 'uy', 'uz', 's', 'rup', 'dxdbup', 'dxdbdn',
```

Comment:
Please don't use ''' for multi-line comments, only for docstrings. Please remove commented code.